src/shared_utilities/FileReader.jl

```diff
@@ -144,10 +146,17 @@ simulation grid. Date-related args (last 3 to FileInfo) are unused for static
 data maps.
 """
 function PrescribedDataStatic(
-    infile_path::String,
+    get_infile_path::Function,
     regrid_dirpath::String,
     varname::String,
+    comms_ctx::ClimaComms.AbstractCommsContext,
 )
+    # Download `infile_path` artifact on root process first to avoid race condition
+    if ClimaComms.iamroot(comms_ctx)
+        infile_path = get_infile_path()
+    end
+    infile_path = get_infile_path()
```

Comment:
Im a little confused here. if not ClimaComms.iamroot(comms_ctx), does anything change? we still get the file path

Reply:
The goal is for the root to download the file first, then other files can try to download it after (since it's already downloaded, they'll just access it instead of downloading). If they all call `get_infile_path()` at the same time (like we had before), some threads try to read the file while another thread is still downloading it, so we get a `No such file or directory` error like [here](https://buildkite.com/clima/climacoupler-longruns/builds/251#018b3fee-be6a-4792-8df6-63daaf906e49)

Also, later on in the function, `infile_path` is used by all processes (not just root), so it has to be defined as a variable on all processes

Now that I think about it some more, I should have a barrier after the `iamroot` if statement, to enforce that all the threads have to wait for the root process to finish downloading. I'll add that right now

Reply:
makes sense! Thanks!